android-DL4JImageRecognitionDemo.md

```diff
@@ -0,0 +1,382 @@
+
+![](images/screen2.png)
+## <a name="head_link1">Setting the Dependencies</a>
+Deeplearning4J applications requires application specific dependencies in the build.gradle file. The Deeplearning library in turn depends on the libraries of ND4J and OpenBLAS, thus these must also be added to the dependencies declaration. Starting with Android Studio 3.0, annotationProcessors need to be defined as well, thus dependencies for either -x86 or -arm processors should be included, depending on your device, if you are working in Android Studio 3.0 or later. Note that both can be include without conflict as is done in the example app.
+```java
+	compile 'com.android.support:appcompat-v7:27.0.2'
+        compile 'com.android.support:design:27.0.2'
+        compile 'org.deeplearning4j:deeplearning4j-nn:0.9.1'
+        compile 'org.nd4j:nd4j-native:0.9.1'
+        compile 'org.nd4j:nd4j-native:0.9.1:android-x86'
+        compile 'org.nd4j:nd4j-native:0.9.1:android-arm'
+        compile 'org.bytedeco.javacpp-presets:systems-platform:1.4'
+        compile 'org.bytedeco.javacpp-presets:openblas:0.2.19-1.3:android-x86'
+        compile 'org.bytedeco.javacpp-presets:openblas:0.2.19-1.3:android-arm'
+        testCompile 'junit:junit:4.12'
+```
+Depending on the combination of dependencies, duplication conflicts can arise that must be handled with exclusions. After adding the above dependencies and the exclusions listed below, sync the Gradle file add additional exclusions if needed. The error message will identify the file path that should be added to the list of exclusions. An example error message with file path: **> More than one file was found with OS independent path 'org/bytedeco/javacpp/ windows-x86_64/msvp120.dll'**
+```java
+packagingOptions {
+ 
+	exclude 'META-INF/DEPENDENCIES'
+	exclude 'META-INF/DEPENDENCIES.txt'
+	exclude 'META-INF/LICENSE'
+	exclude 'META-INF/LICENSE.txt'
+	exclude 'META-INF/license.txt'
+	exclude 'META-INF/NOTICE'
+	exclude 'META-INF/NOTICE.txt'
+	exclude 'META-INF/notice.txt'
+	exclude 'META-INF/INDEX.LIST'
+ 
+	exclude 'org/bytedeco/javacpp/windows-x86/msvcp120.dll'
```

Comment:
Is this still happening with 0.9.1? It shouldn't be...

Reply:
Right again, those exclusions are no long needed. I will remove them from the code.

Reply:
Actually, they might pop up again if we add a dependency deeplearning4j-core. What would be better is to exclude all the "-platform" artifacts like ` exclude group: 'org.bytedeco.javacpp-presets', module: 'opencv-platform' ` and add manually the ones we need like `org.bytedeco.javacpp-presets:opencv:version:android-arm`. That's the kind of thing we'd ideally want a gradle-javacpp plugin do for us...

Reply:
I removed the unnecessary exclusions. There are instructions in that section of all three tutorials that explain how to add  exclusions if any errors arise from duplicate files based on the dependencies used. Does that sound OK?